src/modules/home/components/EventCarousel.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="relative rounded-2xl overflow-hidden h-40 w-120 z-20 shadow">
+    <event-carousel-slide
+      v-for="(event, slideIndex) in testEventProps"
+      :key="event.eventTitle"
+      :slideIndex="slideIndex"
+      :currentIndex="currentIndex"
+      :transitionAnimation="transitionAnimation"
+    >
+      <CardBanner class="w-full" :eventTitle="event.eventTitle" />
+    </event-carousel-slide>
+    <section class="carousel-misc absolute z-10">
+      <div
+        v-for="index in indexRange"
+        :key="index"
+        @click="changeSlide(index)"
+        class="w-1"
+        :class="{
+          'carousel-gap': index !== 3
+        }"
+      >
+        <div
+          class="carousel-dot bg-white"
+          :class="{
+            'h-3': index === currentIndex,
+            'h-2 bg-opacity-30': index !== currentIndex
+          }"
+        ></div>
+      </div>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from "vue";
+import CardBanner from "./CardBanner.vue";
+import EventCarouselSlide from "./EventCarouselSlide.vue";
+import makeRange from "@/commons/utils/range.ts";
+
+export default defineComponent({
+  name: "EventCarousel",
+  components: {
+    CardBanner,
+    EventCarouselSlide
+  },
+  setup() {

```

Comment:
Can we wrap logic about carousel into useCaroisel function file. This will make logic separated from component and enable easier  unit testing in future